bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -25,17 +31,63 @@
  *
  * @author Markus Rathgeb - Initial contribution and API
  * @author Mark Herwege - Added methods to find broadcast address(es)
+ * @author Stefan Triller - Converted to OSGi service with primary ipv4 conf
  */
-public class NetUtil {
+@Component(name = "org.eclipse.smarthome.network", property = { "service.config.description.uri=system:network",
+        "service.config.label=Network Settings", "service.config.category=system" })
+public class NetUtil implements NetworkAddressProvider {
```

Comment:
Sorry, just realised your name choice now: "Provider" has a very special meaning in many contexts in ESH. `NetworkAddressService` would probably a more neutral choice.